src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/IOperationClassWriter.Verifier.cs

```diff
@@ -51,6 +51,15 @@ private bool ModelHasErrors(Tree tree)
                     }
                 }
 
+                foreach (var prop in GetAllGeneratedIOperationProperties(abstractNode))
+                {
+                    if (IsImmutableArray(prop.Type, out _) && prop.Type.Contains("?"))
+                    {
+                        Console.WriteLine($"{abstractNode.Name}.{prop.Name} has nullable IOperation elements. This is not allowed in IOperation and will mess up Children generation.");
```

Comment:
>Console.WriteLine($"{abstractNode.Name}.{prop.Name} has nullable IOperation elements. This is not allowed in IOperation and will mess up Children generation."); [](start = 24, length = 160)

Why not handle the nulls?

Reply:
Simplicity. We don't have any of these today, and not having to do extra null checks makes generating the code a bit simpler. I'd have to be convinced that exposing an array of nullable operations in the first place would be a good idea, but this will ensure that if we ever do have a reason to do so, we don't forget to update the implementation of the enumerator methods.

---
In reply to: [539571031](https://github.com/dotnet/roslyn/pull/49804#discussion_r539571031) [](ancestors = 539571031)